lib/credentials/shared_ini_file_credentials.js

```diff
@@ -199,17 +221,44 @@ AWS.SharedIniFileCredentials = AWS.util.inherit(AWS.Credentials, {
       roleParams.ExternalId = externalId;
     }
 
+    if (mfaSerial) {
+      roleParams.SerialNumber = mfaSerial;
+      self.tokenCodeFn(mfaSerial, function(err, token) {
+        if (err) {
+          self.callRefreshCallbacks(
+            AWS.util.error(
+              new Error('Error fetching MFA token: ' + err),
```

Comment:
Assuming `err` is an Error object, I think it'd be fair to update this to use `err.message` instead of `err`.